chore(cli): rename executable to miden-vm to disambiguate

bitwalker opened the pull request:
This PR renames the `miden` executable to `miden-vm`, both to disambiguate with the `miden-client` executable (which is also currently named `miden`, and which I will be making a similar PR to change), and to free up the `miden` name for our AIO (all-in-one) `miden` CLI interface (installed by `midenup`).

I believe I've updated all of the tests and documentation to reflect the new name, but let me know if there are other places that need to be updated.

I've also marked this as a breaking change in the CHANGELOG, since even if the new `miden` CLI was installed at the same time this one was renamed, the interface will almost certainly be different, so downstream users will need to be aware of the change.

bitwalker commented on docs/src/intro/usage.md:
````diff
@@ -20,7 +20,7 @@ To compile Miden VM into a binary, we have a [Makefile](https://www.gnu.org/soft
 ```
 
-This will place an optimized, multi-threaded `miden` executable into the `./target/optimized` directory. It is equivalent to executing:
+This will place an optimized, multi-threaded `miden-vm` executable into the `./target/optimized` directory. It is equivalent to executing:
````
Thanks! I'll get those fixed up, unfortunately very awkward to search for 😅

Al-Kindi-0 reviewed (approved):
Looks good to me!

A few places that I believe that are still missing (the ones with "miden program" are more from a stylistic pov though):

- debugger.md
- #[command(about = "Run a miden program")] in miden-vm/src/cli/run.rs
- [command(about = "Verify a miden program")]  in miden-vm/src/cli/verify.rs
- [command(about = "Debug a miden program")] in miden-vm/src/cli/debug/mod.rs
- #[command(about = "Analyze a miden program")] in miden-vm/src/tools/mod.rs
- `miden/tests/integration/cli/cli_test.rs::test_debug_adv_stack` in the "test_parsing_debug_advice_stack()" test
- miden-vm/src/cli/repl/mod.rs has a few "miden" occurences in the comments e.g., "miden program"

bitwalker commented:
> * #[command(about = "Run a miden program")] in miden-vm/src/cli/run.rs
> * #[command(about = "Verify a miden program")]  in miden-vm/src/cli/verify.rs
> * #[command(about = "Debug a miden program")] in miden-vm/src/cli/debug/mod.rs
> * #[command(about = "Analyze a miden program")] in miden-vm/src/tools/mod.rs

@Al-Kindi-0 I think in general, phrases like _a Miden program_ would remain unchanged, but I have fixed incorrect capitalization in a few places to better distinguish between Miden as a project and the `miden` executable. All of your other flags though I've corrected, thanks for catching those!
